packages/loader/container-utils/src/error.ts

```diff
@@ -180,11 +185,34 @@ export class DataProcessingError extends LoggingError implements IErrorBase, IFl
     }
 }
 
-export const extractSafePropertiesFromMessage = (message: ISequencedDocumentMessage) => ({
-    messageClientId: message.clientId,
-    messageSequenceNumber: message.sequenceNumber,
-    messageClientSequenceNumber: message.clientSequenceNumber,
-    messageReferenceSequenceNumber: message.referenceSequenceNumber,
-    messageMinimumSequenceNumber: message.minimumSequenceNumber,
-    messageTimestamp: message.timestamp,
-});
+export const extractSafePropertiesFromMessage = (message: ISequencedDocumentMessage) => {
+    const baseTelemetry = {
+        messageClientId: message.clientId,
+        messageSequenceNumber: message.sequenceNumber,
+        messageClientSequenceNumber: message.clientSequenceNumber,
+        messageReferenceSequenceNumber: message.referenceSequenceNumber,
+        messageMinimumSequenceNumber: message.minimumSequenceNumber,
+        messageTimestamp: message.timestamp,
+    };
+
+    switch (message.type) {
+        case MessageType.ClientJoin: {
+            const systemJoinMessage = message as ISequencedDocumentSystemMessage;
+            const join = JSON.parse(systemJoinMessage.data) as IClientJoin;
+            return {
+                ...baseTelemetry,
+                joinMessageClientId: join.clientId,
```

Comment:
This could be just `messageClientId`, but generally that field indicates client that generated an op. Here we are more explicit about this being a specific system op. If we add further fields here, they would be joinMessage*

Curious why we do not surface `message.type` ? Are there PII concerns around implying op type ?

Reply:
I doubt it... if so we could tag it as `CodeArtifact` though